### PR TITLE
Allow script/style before html

### PR DESCRIPTION
**Pull request by natemoo-re**
## Changes

- Allows `script` nodes before the `html` node.
- Closes https://github.com/withastro/compiler/issues/660
- This was an edge case in our parser that was causing the actual `html` node to be ignored

## Testing

Wasm test added

## Docs

Bug fix only

### CODE REVIEW

File: packages/compiler/test/basic/script-before-html.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import { test } from 'uvu';
+import * as assert from 'uvu/assert';
+import { transform } from '@astrojs/compiler';
+
+const FIXTURE = `
+<script>
+  // anything ...
+</script>
+
+<html lang="de">
```
Comment (jasikpark) [inline packages/compiler/test/basic/script-before-html.ts:10]: maybe worth adding custom DOCTYPE in the test here as well?

**Review by jasikpark — COMMENTED** [review]
don't fully understand the parser code, but the test looks like it validates the bugfix 👍

**Review by jasikpark — APPROVED** [review]
